samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/removingproductitem/RemoveProductItemFromShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package io.eventdriven.ecommerce.shoppingcarts.removingproductitem;
+
+import io.eventdriven.ecommerce.shoppingcarts.Events;
+import io.eventdriven.ecommerce.shoppingcarts.ShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.productitems.PricedProductItem;
+
+import java.util.UUID;
+
+public record RemoveProductItemFromShoppingCart(
+  UUID shoppingCartId,
+  PricedProductItem productItem,
+  Long expectedVersion
+) {
+  public static RemoveProductItemFromShoppingCart of(UUID cartId, PricedProductItem productItem, Long expectedVersion) {
```

Comment:
I'm not sure if the of method it's good in this case. I also like factory methods, but when you're using records, the consturctor is always public, so it's possible to use constructor instead of method (in normal class you can make constructor private). So in this case maybe better do the validation in the constructor?